plugins/hls-cabal-plugin/src/Ide/Plugin/Cabal.hs

```diff
@@ -277,6 +280,38 @@ fieldSuggestCodeAction recorder ide _ (CodeActionParams _ _ (TextDocumentIdentif
       let completionTexts = fmap (^. JL.label) completions
       pure $ FieldSuggest.fieldErrorAction uri fieldName completionTexts _range
 
+-- | CodeActions for going to definitions.
+--
+-- Provides a CodeAction for going to a definition when clicking on an identifier.
+-- The definition is found by traversing the sections and comparing their name to
+-- the clicked identifier.
+--
+-- TODO: Support more definitions than sections.
+gotoDefinition :: PluginMethodHandler IdeState LSP.Method_TextDocumentDefinition
+gotoDefinition ideState _ msgParam = do
+    case uriToFilePath' uri of
+      Nothing ->
+        pure $ InR $ InR Null
+      Just filePath -> do
```

Comment:
Prefer `getNormalizedFilePathE` which takes care of the validation.

```suggestion
    nfp <- getNormalizedFilePathE uri
```

Then use `fromNormalizedFilePath` if you need the underlying `FilePath`.